app/src/Flashcards/EmptySides.js

```diff
@@ -54,7 +59,12 @@ const EmptySides = ({
   }, [id, onClickDeleteCard]);
   return (
     <>
-      <DictionarySearch />
+      {location.pathname === "/new-card" && (
+        <DictionarySearch
+          definition={definition}
+          setDefinition={setDefinition}
+        />
+      )}
```

Comment:
Instead of using the location to check the current path, why not just do the following inside `CreateCard`?

```
<CreateCard>
  <EmptySides />
  <DictionarySearch
    definition={definition}
    setDefinition={setDefinition}
  />
</CreateCard>
```

since `CreateCard` is only rendered with the `location` is `/new-card`, this should work the same way without creating a dependency on the URL inside `EmptySides`.

Reply:
I can do that, but I feel like using the location is more convenient for me because if I place DictionarySearch component in the CreateCard component I would need to move "front", "back", "setFront" and "setBack" states to the CreateCard component to pass them between EmptySides sibling and the DictionatySeatch, but I also use those in the EditCard component. So it looks more complicated to me.